src/cpp/session/SessionHttpMethods.cpp

```diff
@@ -827,6 +827,12 @@ void handleConnection(boost::shared_ptr<HttpConnection> ptrConnection,
             if (console_input::executing())
                rstudio::r::exec::setInterruptsPending(true);
          }
+         
+         // ping
+         else if (jsonRpcRequest.method == kPing)
```

Comment:
Because "ping" is a high-priority request, I think it makes sense to process it immediately rather than queueing it like other RPC methods.